### ISSUE TITLE
Release tag message flag

### ISSUE THREAD

**Issue by Zoramite** [nvie/gitflow#50]
When attempting to use the message flag for finishing a release I am getting an error.

Here is what I attempted to run:

```
git flow release finish 0.1.3 -m 'Releasing version 0.1.3'
```

And here is what it returns:

```
flags:WARN getopt: invalid option -- 'm'
 -- 'release' 'finish' '0.1.3' 'Releasing version 0.1.3'
flags:FATAL unable to parse provided options with getopt.
```

Running the latest from the `develop` branch.

**Comment by nvie** [comment]
Closing this, as this has the same cause (under the hood) as issue #28.

**Comment by Zoramite** [comment]
I no longer get the fatal flag message but now if I use the -m flag it ignores it and opens the default editor.

```
git flow release finish '0.1.0' -m 'Testing a release'
```

Using Ubuntu 10.10 with latest develop branch

**Comment by nvie** [comment]
Yeah, the flags should come _before_ the arguments, Unix style. This is an shFlags limitation, I'm afraid.

It might be a an option to error on superfluous arguments (as the -m and the message are treated), but this leads to a whole lot of other issues, I'm afraid. Best for now to stick to good old Unix habits.. :(

Cheers,
Vincent

**Comment by Zoramite** [comment]
Interesting, I hadn't made that connection before with the flags.

I tried this:

```
git flow release finish -m 'Testing tagging' '0.1.0'
```

But I get this error now:

```
fatal: Failed to resolve '0.1.0' as a valid ref.
Tagging failed. Please run finish again to retry.
```

Running `git branch`:

```
  develop
  master
* release/0.1.0
```

Running the command without the `-m` flag still works.

**Comment by nvie** [comment]
Hmm. I've tested things here on my Mac, but the only thing I get is this:

```
flags:FATAL the available getopt does not support spaces in options
```

on the space in the quoted `-m` flag value, which scares me even more :)

What OS are you running?

**Comment by Zoramite** [comment]
I'm using Ubuntu 10.10 with the latest from the develop branch.

**Comment by elight** [comment]
Is this related? I'm seeing this on OS X:

```
➜  ~rvm_rvmrc_cwd git:(hotfix/fix_loc_search_500_on_staging_and_prod) ✗ git flow hotfix finish fix_loc_search_500_on_staging_and_prod
Switched to branch 'master'
Merge made by recursive.
 app/controllers/api/locations_controller.rb |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)
fatal: no tag message?
Tagging failed. Please run finish again to retry. 
```

**Comment by andreareginato** [comment]
I'm having the same behavior of elight. Right now I have to use git commands to merge and tag, because it do not work.

**Comment by xiaods** [comment]
also come cross this msg occur:

[xds2000@localhost demo](develop)$ git flow release finish 'v0.1.0' -m 'Testing a release'
fatal: no tag message?
Tagging failed. Please run finish again to retry.

**Comment by nvie** [comment]
@xiaods: Yeah, like said, you should use the `-m` flag _before_ the branch name, UNIX style.

**Comment by xiaods** [comment]
@nvie, i have review this thread list, see our's comments. i suggesting print some hits on it?

**Comment by lesliev** [comment]
@nvie can you pull these changes? They work for me. 
Thanks @Zoramite, I can now do: git flow release finish -m "$MSG" $REL

**Comment by unluckypixie** [comment]
Hi,

This issue is really bugging me and seems to be fixed yet when I pull from git hub the latest changes seem to be Dec 06 2011? So I assume I've not got the fix discussed above done in late december ...

Has the github respository not been updated?  Or am I doing something wrong to get the latest version?

Ta.

**Comment by petervanderdoes** [comment]
It has not been updated since Dec 06 2011

**Comment by lesliev** [comment]
I applied the fix myself, since its just two lines to edit: https://github.com/nvie/gitflow/pull/160/files
- or you could just use @Zoramite's fork.

Perhaps @nvie is otherwise engaged, there do seem to be a large number of forks for Gitflow.

**Comment by unluckypixie** [comment]
Thanks, that's great, I'll apply the fix myself.

**Comment by nvie** [comment]
Closing all issues as I'm archiving this repo. git-flow lives on at [git-flow-next](https://github.com/gittower/git-flow-next) if you're still using it. Thanks for reporting!
